src/js/kits/model-kit-select/select.js

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+
+const Select = (props) => {
+    const { fetchOptions, onChange } = props;
+    const [options, setOptions] = useState([{value: '', label: '(loading...)'}]);
+
+    useEffect(() => {
+        fetchOptions()
+            .then(options => setOptions(options))
+            .catch(console.error);
+    }, []);
+
+    return (
+        <select onChange={ (event) => onChange(event.target.value) } className="form-control">
```

Comment:
I'd return the whole event here: `onChange={onChange}`. That way you can pass `name` prop to select and do smth like this from wherever it's used:
```
const onChange = event => {
  const { name, target } = event.target; 
  updateSearchParam(name, value);
}

<BrandSelect onChange={ onChange } />
```
to avoid stuff like `onChange={ (pk) => updateSearchParam('brand', pk) }`.

Reply:
handled it differently by making `AsyncSelect` more generic. Wherever it's being called, I know the relevant name in the parent component, and that's the only place that it's relevant. I'm only interested in the value from the select.